lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -868,6 +866,31 @@ def estimate_diffusion(self, method, *args, min_length=None, **kwargs):
 
         return [k.estimate_diffusion(method, *args, **kwargs) for k in filtered_tracks]
 
+    def ensemble_diffusion(self, method):
+        """Determine ensemble based diffusion estimates.
+
+        Determines ensemble based diffusion estimates for the entire group of KymoTracks. This
+        method assumes that all tracks experience the same diffusion and computes an averaged
+        diffusion estimate.
+
+        Parameters
+        ----------
+        method : {"cve"}
+            - "cve" : Covariance based estimator. Optimal if SNR > 1. Ensemble average is
+              determined by determining the weighted average of the individual track estimates. The
+              standard error is computed by determining the weighted average of the associated
+              standard errors for each estimate (Equation 57 and 58 from Vestergaard [7]_). See
+              :meth:`KymoTrack.estimate_diffusion` for more detailed information and references.
+
+        References
+        ----------
+        .. [7] Vestergaard, C. L., Blainey, P. C., & Flyvbjerg, H. (2014). Optimal estimation of
+               diffusion coefficients from single-particle trajectories. Physical Review E, 89(2),
+               022726.
+        """
+        if method == "cve":
+            return ensemble_cve(self)
```

Comment:
maybe `else` throw? seems overkill maybe but I could imagine someone having a typo or using `"CVE"` in which case this will just silently do nothing right?